File: src/main/java/com/example/demo/entity/Drink.java
package com.example.demo.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
public class Drink {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int iddrink;
	private String drinkname, drinkcategory, drinkdescription;
	private double drinkprice;
	private byte[] drinkimage;
}

File: lib/models/drink.dart
import 'dart:typed_data';
import 'dart:convert';

class Drink {
  final int iddrink;
  final String drinkname;
  final String drinkdescription;
  final String drinkcategory;
  final double drinkprice;
  final Uint8List drinkimage;

  Drink({
    required this.iddrink,
    required this.drinkname,
    required this.drinkdescription,
    required this.drinkcategory,
    required this.drinkprice,
    required this.drinkimage,
  });

  factory Drink.fromJson(Map<String, dynamic> json) {
    return Drink(
      iddrink: json['iddrink'],
      drinkname: json['drinkname'],
      drinkdescription: json['drinkdescription'],
      drinkcategory: json['drinkcategory'],
      drinkprice: json['drinkprice'],
      drinkimage: json['drinkimage'] != null
          ? base64Decode(json['drinkimage'])
          : Uint8List(0), 
    );
  }
}
